openjpa-persistence/src/main/java/org/apache/openjpa/persistence/meta/AnnotationProcessor6.java

```diff
@@ -311,10 +314,26 @@ private void annotate(SourceCode source, String originalClass) {
         SourceCode.Class cls = source.getTopLevelClass();
         cls.addAnnotation(StaticMetamodel.class.getName())
             .addArgument("value", originalClass + ".class", false);
-        if (generatedSourceVersion >= 6) {
-            cls.addAnnotation(Generated.class.getName())
-            .addArgument("value", this.getClass().getName())
-            .addArgument("date", new Date().toString());
+        if ("false".equals(this.addGeneratedOption)) {
+            return;
+        }
+
+        if ("force".equals(this.addGeneratedOption)) {
+            cls.addAnnotation(javax.annotation.Generated.class.getName())
+                    .addArgument("value", this.getClass().getName())
+                    .addArgument("date", new Date().toString());
```

Comment:
know it was "before" but can be worth storing the date in init if needed (once the forName moved there it will be easy to know)

Reply:
Will do!